### ISSUE TITLE
Error setting up project with ceramic init

### ISSUE THREAD

**Issue by anissen** [ceramic-engine/ceramic#65]
I'm following the guide at https://jeremyfa.com/getting-started-with-ceramic/.

After installing ceramic I get an error trying to initialize a project.
`➜  ceramic-test ceramic init --name myproject`
```
Project created at path: /Users/anders/projects/misc/ceramic-test/myproject
dyld: Library not loaded: @rpath/libneko.2.dylib
  Referenced from: /Users/anders/projects/misc/ceramic-test/ceramic/tools/../git/haxe-binary/mac/haxe/haxelib
  Reason: image not found
/Users/anders/projects/misc/ceramic-test/ceramic/tools/haxelib: line 11: 91489 Abort trap: 6           "${0%/*}"/../git/haxe-binary/mac/haxe/haxelib "$@"
Error: ENOENT: no such file or directory, open '/Users/anders/projects/misc/ceramic-test/myproject/.haxelib/hxcpp/4,2,1/toolchain/android-toolchain-clang.xml'
    at Object.openSync (fs.js:443:3)
    at Object.readFileSync (fs.js:343:35)
    at Function.tools_Helpers.checkProjectHaxelibSetup (/Users/anders/projects/misc/ceramic-test/ceramic/plugins/clay/file:/Users/runner/work/ceramic/ceramic/tools/src/tools/Helpers.hx:374:37)
    at tools_tasks_Setup.run (/Users/anders/projects/misc/ceramic-test/ceramic/plugins/clay/file:/Users/runner/work/ceramic/ceramic/tools/src/tools/tasks/Setup.hx:26:9)
    at Function.tools_Tools.run (/Users/anders/projects/misc/ceramic-test/ceramic/tools/file:/Users/runner/work/ceramic/ceramic/tools/src/tools/Tools.hx:259:17)
    at /Users/anders/projects/misc/ceramic-test/ceramic/tools/file:/Users/runner/work/ceramic/ceramic/tools/src/tools/Tools.hx:30:13
```

I get the same error (`dyld: Library not loaded: @rpath/libneko.2.dylib`) if I try to run `/Users/anders/projects/misc/ceramic-test/ceramic/git/haxe-binary/mac/haxe/./haxelib` directly.

I'm running macOS and have Haxe 4.2.2 installed (through Lix).

**Comment by jeremyfa** [comment]
ceramic is not compatible with lix at the moment. You need to have neko available globally (easiest way to do that is to install haxe globally with official installer as stated in the guide).

lix support might be considered in a future release though, but is not a priority for me as I'm not using it personally.

**Comment by jeremyfa** [comment]
Well, I might have a fix soon to make it work in your setup though!

**Comment by anissen** [comment]
Thanks for the quick replies!

I do have neko installed globally:
```
➜  which neko
/usr/local/bin/neko
➜  neko
NekoVM 2.3.0 (c)2005-2017 Haxe Foundation
  Usage : neko <file>
```

I can find the `libneko.2.dylib` library in two locations:
```
/Users/anders/haxe/neko/libneko.2.dylib
/Users/anders/projects/misc/ceramic-test/ceramic/git/haxe-binary/mac/neko/libneko.2.dylib
```

I would expect ceramic to work regardless of how haxe was installed (e.g. through lix, npm, homebrew or the official installer). Seeing as neko is included with ceramic, I would expect it to look for neko in the ceramic directory rather than the "officiel" haxe installation.

**Comment by jeremyfa** [comment]
> I would expect ceramic to work regardless of how haxe was installed (e.g. through lix, npm, homebrew or the official installer). Seeing as neko is included with ceramic, I would expect it to look for neko in the ceramic directory rather than the "officiel" haxe installation.

That's what I would like but unfortunately when calling `haxe`, it may use `haxelib` internally without propagating environment variables to let `haxelib` find neko where it is bundled with ceramic and as far as I know there is nothing I can do about it for now, except ensuring libneko can be found on default paths.

I'm testing something to let ceramic automatically create symbolic links of libneko & similar into `/usr/local/lib` if none exist to make haxe/haxelib work properly no matter what, and it may solve your problem. I wish I didn't have to do that though.

**Comment by jeremyfa** [comment]
Posted an issue on haxe repo about this https://github.com/HaxeFoundation/haxe/issues/10297

**Comment by jeremyfa** [comment]
I published a new release of ceramic (v0.7.11). You can update ceramic via `haxelib run ceramic setup` and try to initialize a new project. Hopefully this time it will work. Let me know!

**Comment by anissen** [comment]
I tried updating my existing ceramic installation, but got the following error.

```
Detected existing ceramic version v0.7.10 (/Users/anders/projects/misc/ceramic-test/ceramic)
Update ceramic to v0.7.11? (y/n)
y
Downloading ceramic-mac.zip...
Download complete : 621 bytes in 0.3s (1.8KB/s)
Download complete : 185957932 bytes in 18.7s (9662KB/s)
Called from ? line 1
Called from Main.hx line 88
Called from Main.hx line 153
Called from Main.hx line 361
Called from Main.hx line 361
Called from Main.hx line 361
Called from Main.hx line 361
Called from Main.hx line 361
Called from Main.hx line 361
Called from Main.hx line 361
Called from Main.hx line 361
Called from Main.hx line 361
Called from Main.hx line 367
Called from /usr/local/lib/haxe/std/neko/_std/sys/FileSystem.hx line 91
Uncaught exception - std@sys_remove_dir
```

However, I then manually removed and reinstalled ceramic... and it works! 🎉 

I'm hoping to find some time to play with ceramic this weekend. Thanks for your help! 🍰

**Comment by anissen** [comment]
An update:

I tried to run the project without the ceramic extension I got an error in tasks.json in `"type": "ceramic",` (saying that only "haxe" was a valid type). After installing the ceramic extension, ceramic seems to work great with vscode. I've successfully run the scaffolded project on web, macOS and iPadOS targets without problems.

So far ceramic seems quite polished with a lot of thought put into the ecosystem and development process. Looking forward to exploring it more 👍🏻
